pkg/providers/openai/chat.go

```diff
@@ -4,232 +4,147 @@ import (
-		Messages:         messages,
-		Temperature:      0.8,
-		TopP:             1,
-		MaxTokens:        100,
-		N:                1,
-		StopWords:        []string{},
-		Stream:           false,
-		FrequencyPenalty: 0,
-		PresencePenalty:  0,
-		LogitBias:        nil,
-		User:             nil,
-		Seed:             nil,
-		Tools:            []string{},
-		ToolChoice:       nil,
-		ResponseFormat:   nil,
+// NewChatRequestFromConfig fills the struct from the config. Not using reflection because of performance penalty it gives
+func NewChatRequestFromConfig(cfg *Config) *ChatRequest {
+	return &ChatRequest{
+		Model:            cfg.Model,
+		Temperature:      cfg.DefaultParams.Temperature,
+		TopP:             cfg.DefaultParams.TopP,
+		MaxTokens:        cfg.DefaultParams.MaxTokens,
+		N:                cfg.DefaultParams.N,
+		StopWords:        cfg.DefaultParams.StopWords,
+		Stream:           false, // unsupported right now
+		FrequencyPenalty: cfg.DefaultParams.FrequencyPenalty,
+		PresencePenalty:  cfg.DefaultParams.PresencePenalty,
+		LogitBias:        cfg.DefaultParams.LogitBias,
+		User:             cfg.DefaultParams.User,
+		Seed:             cfg.DefaultParams.Seed,
+		Tools:            cfg.DefaultParams.Tools,
+		ToolChoice:       cfg.DefaultParams.ToolChoice,
+		ResponseFormat:   cfg.DefaultParams.ResponseFormat,
 	}
+}
 
-	chatRequestValue := reflect.ValueOf(chatRequest).Elem()
-	chatRequestType := chatRequestValue.Type()
-
-	for i := 0; i < chatRequestValue.NumField(); i++ {
-		jsonTags := strings.Split(chatRequestType.Field(i).Tag.Get("json"), ",")
-		jsonTag := jsonTags[0]
+func NewChatMessagesFromUnifiedRequest(request *schemas.UnifiedChatRequest) []ChatMessage {
+	messages := make([]ChatMessage, 0, len(request.MessageHistory)+1)
 
-		if value, ok := defaultParams[jsonTag]; ok {
-			fieldValue := chatRequestValue.Field(i)
-			fieldValue.Set(reflect.ValueOf(value))
-		}
+	// Add items from messageHistory first and the new chat message last
+	for _, message := range request.MessageHistory {
+		messages = append(messages, ChatMessage{Role: message.Role, Content: message.Content})
 	}
 
-	// c.Telemetry.Logger.Info("chatRequest created", zap.Any("chatRequest body", chatRequest))
+	messages = append(messages, ChatMessage{Role: request.Message.Role, Content: request.Message.Content})
 
-	return chatRequest
+	return messages
 }
 
-// CreateChatResponse creates chat Response.
-func (c *ProviderClient) CreateChatResponse(ctx context.Context, r *ChatRequest, u *providers.UnifiedAPIData) (*ChatResponse, error) {
-	_ = ctx // keep this for future use
+// Chat sends a chat request to the specified OpenAI model.
+func (c *Client) Chat(ctx context.Context, request *schemas.UnifiedChatRequest) (*schemas.UnifiedChatResponse, error) {
+	// Create a new chat request
+	chatRequest := c.createChatRequestSchema(request)
 
-	resp, err := c.createChatHTTP(r, u)
+	// TODO: this is suspicious we do zero remapping of OpenAI response and send it back as is.
```

Comment:
No, I need to update per our discussion of unified response. Will be working on mapping this week.

Reply:
That's good to know then I'm leaving this spot without changes